internal/module/services/version_manager/version_manager.go

```diff
@@ -235,6 +236,27 @@ func (s *VersionManagerService) ListModuleVersions(ctx context.Context, request
 			grpcResponse.Versions = append(grpcResponse.Versions, moduleVersion.Version)
 		}
 	}
+	var semverList versions.List
+	for _, moduleVersion := range grpcResponse.Versions {
+		parsedVersion, err := versions.ParseVersion(moduleVersion)
+
+		if err != nil {
+			log.Printf("Skipping invalid semantic version: %v", moduleVersion)
+		} else {
+			if (parsedVersion.GreaterThan(versions.MustParseVersion("0.0.0"))) {
```

Comment:
This can't be done there. The filtering must be as part of the browse service. the version manager services does provide information to both the browse and and the rest API used by terraform CLI. the module.v1 API needs to list all valid version while the browse should only list version that are official release. 

As such the filtering can't be done at the low level. Provided ListModule Versions is used across both browse and module.v1 (https://github.com/dkPranav/terrarium/blob/58499c75ccfc6d60935fa8162eed44a0ca19979d/internal/restapi/modules/v1/handler.go#L101C35-L101C53) then this filtering can't be here.

Reply:
Updated the browse handler to filter out the developmental version. @odelanno